pennylane/qchem/tapering.py

```diff
@@ -579,39 +590,42 @@ def _build_generator(operation, wire_order, op_gen=None):
     if op_gen is None:
         if operation.num_params < 1:  # Non-parameterized gates
             gen_mat = 1j * scipy.linalg.logm(qml.matrix(operation, wire_order=wire_order))
-            op_gen = qml.pauli_decompose(gen_mat, wire_order=wire_order, hide_identity=True)
-            op_gen = qml.simplify(op_gen)
-            if op_gen.ops[0].label() == qml.Identity(wires=[wire_order[0]]).label():
-                op_gen -= qml.Hamiltonian([op_gen.coeffs[0]], [qml.Identity(wires=wire_order[0])])
-        else:  # Single-parameter gates
-            try:
-                op_gen = qml.generator(operation, "hamiltonian")
-
-            except ValueError as exc:
-                raise NotImplementedError(
-                    f"Generator for {operation} is not implemented, please provide it with 'op_gen' args."
-                ) from exc
-    else:  # check that user-provided generator is correct
-        if not isinstance(op_gen, qml.Hamiltonian):
-            raise ValueError(
-                f"Generator for the operation needs to be a qml.Hamiltonian, but got {type(op_gen)}."
+            op_gen = qml.pauli_decompose(
+                gen_mat, wire_order=wire_order, hide_identity=True, pauli=True
             )
-        coeffs = 1.0
+            op_gen.simplify()  # simplify pauli sentence
+            op_gen.pop(
+                qml.pauli.PauliWord({})
+            )  # remove identity term as this is just a global phase
+
+            if active_new_opmath():
```

Comment:
How does one use `taper_operation` with opmath? I couldn't taper `qml.SingleExcitation` gate after using `enable_new_opmath`. 🤔

Reply:
Its not currently supported because of `is_commuting()`